### PR TITLE
Add a new option for address sanitization with assertion message  (unimplemented feature yet)

### PR DESCRIPTION
**Pull request by inho9606**
Added a new option -iree-llvm-address-sanitizer to iree-translate, but it's feature is not implemented yet.
Assertion error occurs when the new option is turned on from debug mode. On release mode, it ignores the option since it is not implemented yet.

### CODE REVIEW

File: iree/compiler/Dialect/HAL/Target/LLVM/LLVMIRPasses.cpp [hunk 1]
```diff
@@ -66,6 +66,8 @@ std::unique_ptr<llvm::TargetMachine> createTargetMachine(
 LogicalResult runLLVMIRPasses(const LLVMTargetOptions &options,
                               llvm::TargetMachine *machine,
                               llvm::Module *module) {
+  if(options.addressSanitizer)
```
Comment (benvanik) [inline iree/compiler/Dialect/HAL/Target/LLVM/LLVMIRPasses.cpp:69]: style: (almost) always use braces for multi-line statements in IREE code:
https://google.github.io/styleguide/cppguide.html#Conditionals

File: iree/compiler/Dialect/HAL/Target/LLVM/LLVMIRPasses.cpp [hunk 2]
```diff
@@ -66,6 +66,8 @@ std::unique_ptr<llvm::TargetMachine> createTargetMachine(
...
+    assert(false && "Address Sanitization feature has not been implemented yet.");
```
Comment (benvanik) [inline iree/compiler/Dialect/HAL/Target/LLVM/LLVMIRPasses.cpp:70]: avoid assert for expected conditions - it is generally not useful (a lot of people build and run in release mode with asserts disabled) and really difficult to debug. MLIR has `emitError`/`return failure()` to gracefully handle this with good error messages.

File: iree/compiler/Dialect/HAL/Target/LLVM/LLVMTargetOptions.cpp [hunk 3]
```diff
@@ -107,6 +107,13 @@ LLVMTargetOptions getLLVMTargetOptionsFromFlags() {
       llvm::cl::init(llvmTargetOptions.debugSymbols));
   llvmTargetOptions.debugSymbols = clDebugSymbols;
 
+  static llvm::cl::opt<bool> clAddressSanitizer(
+      "iree-llvm-address-sanitizer",
+      llvm::cl::desc("Turn on address sanitization to detect memory violations"),
+      llvm::cl::init(llvmTargetOptions.addressSanitizer));
+//  if(clAddressSanitizer)
```
Comment (benvanik) [inline iree/compiler/Dialect/HAL/Target/LLVM/LLVMTargetOptions.cpp:114]: Please self-review before sending PRs to make sure you don't leave stray whitespace changes and comments :)

**Review by benvanik — CHANGES_REQUESTED** [review]
Thanks for looking into this! We'll want to wait to start merging this until the feature is working - things like this can get particularly nasty when dealing with various platforms/compilers and will likely change a lot while trying to get it all settled. I've left some review comments to help for when that time comes, but for now we'll need to wait. Please let us know how it goes!

**Comment by inho9606** [comment]
Thanks for the helpful comments! Sorry to create another PR after your comments. I missed your comments.

**Comment by benvanik** [comment]
No problem!

**Pull request closed without merging**